app/uk/gov/hmrc/selfservicetimetopay/controllers/EligibilityController.scala

```diff
@@ -69,13 +69,15 @@ class EligibilityController @Inject() (val messagesApi: play.api.i18n.MessagesAp
 
   def submitSignInQuestion: Action[AnyContent] = Action.async { implicit request =>
     sessionCache.get.map {
-      case Some(TTPSubmission(_, _, _, _, `validTypeOfTax`, `validExistingTTP`, cd, _, _, _)) =>
+      case Some(tpp:TTPSubmission)if tpp.eligibilityExistingTtp == `validExistingTTP` && tpp.eligibilityTypeOfTax == `validTypeOfTax`  =>
         EligibilityForm.signInQuestionForm.bindFromRequest().fold(
           formWithErrors => BadRequest(sign_in_question(formWithErrors)),
           {
-            case SignInQuestion(Some(true)) => Redirect(routes.ArrangementController.determineMisalignment())
+            case SignInQuestion(Some(true)) =>
+              if(tpp.calculatorData.debits.nonEmpty) sessionCache.put(tpp.copy(calculatorData = CalculatorInput.initial))
```

Comment:
Put method returns a future so might not be completed in time before redirect so probably best to flatMap on the initial future and then map on this one